src/btree/bt_misc.c

```diff
@@ -139,3 +139,172 @@ __wt_page_type_string(u_int type) WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
     }
     /* NOTREACHED */
 }
+
+/*
+ * __wt_page_is_root --
+ *     Returns true if the page is a root page.
+ */
+static WT_INLINE bool
+__wt_page_is_root(WT_PAGE *page)
+{
+    return (page->pg_intl_parent_ref == NULL || page->pg_intl_parent_ref->home == NULL);
+}
+
+/*
+ * __wt_page_npos --
+ *     Get the page's normalized position in the tree.
+ */
+double
+__wt_page_npos(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    WT_PAGE *page, *parent_page;
+    WT_REF *self_ref;
+    double npos;
+    uint32_t idx, entries;
+
+    npos = 0.5;
+    page = ref->home;
+    if (!page)
+        return (npos);
+
+    for (;;) {
+        WT_ASSERT(session, page != NULL);
+        WT_ASSERT(session, WT_PAGE_IS_INTERNAL(page));
+
+        self_ref = page->pg_intl_parent_ref; /* The field name is deceptive */
```

Comment:
`self_ref` is a bit confusing. I don't have any good suggestions, but maybe `ref_on_parent_page`?